packages/@uppy/aws-s3-multipart/src/index.js

```diff
@@ -663,7 +670,9 @@ export default class AwsS3Multipart extends BasePlugin {
 
         // NOTE This must be allowed by CORS.
         const etag = ev.target.getResponseHeader('ETag')
-        const location = ev.target.getResponseHeader('Location')
+        const location = method.toUpperCase() === 'PUT'
```

Comment:
Sorry to be annoying, but maybe a comment as to why this is needed? Why are removing metadata from URL on PUT?